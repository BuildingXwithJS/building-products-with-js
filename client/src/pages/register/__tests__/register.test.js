// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import RegisterPage, {Register} from '../index';

// create mock store
const mockStore = configureMockStore();

test('# Register page wrapper', () => {
  const store = mockStore({auth: {redirectToLogin() {}}});
  const wrapper = shallow(<RegisterPage store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Register page', () => {
  const newLogin = 'login123';
  const newPass = 'pwd123';
  const navToLogin = () => expect(true).toBeTruthy();
  const redirectToLogin = 'true';
  const onRegisterClick = ({login, password, passwordRepeat}) => {
    expect(login).toBe(newLogin);
    expect(password).toBe(newPass);
    expect(passwordRepeat).toBe(newPass);
  };

  const component = (
    <Register
      onRegisterClick={onRegisterClick}
      navToLogin={navToLogin}
      redirectToLogin={redirectToLogin}
    />
  );

  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // mount for testing
  const app = mount(component);
  // set new login, pass and remember
  app.find('#inputUsername').getDOMNode().value = newLogin;
  app.find('#inputPassword').getDOMNode().value = newPass;
  app.find('#inputPasswordRepeat').getDOMNode().value = newPass;
  // click login button
  app.find('button').simulate('click');
});
