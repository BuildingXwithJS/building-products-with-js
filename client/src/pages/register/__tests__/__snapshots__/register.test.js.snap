// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Register page 1`] = `
<div
  className="jumbotron"
>
  <h2>
    Experts portal:
  </h2>
  <p>
    Please register. Or 
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/login"
    >
      login
    </Link>
  </p>
  <form>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputUsername"
      >
        Username:
      </label>
      <input
        className="form-control"
        id="inputUsername"
        placeholder="Username"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputPassword"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPassword"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="inputPasswordRepeat"
      >
        Password
      </label>
      <input
        className="form-control"
        id="inputPasswordRepeat"
        placeholder="Repeat password"
        type="password"
      />
    </div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`# Register page wrapper 1`] = `
<Register
  navToLogin={[Function]}
  onRegisterClick={[Function]}
  redirectToLogin={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
